diff --git a/supervisor_stdout.py b/supervisor_stdout.py
index 34cfff9..e7b1b5a 100644
--- a/supervisor_stdout.py
+++ b/supervisor_stdout.py
@@ -1,3 +1,4 @@
+from __future__ import print_function
 import sys
 
 def write_stdout(s):
@@ -19,7 +20,9 @@ def main():
 def event_handler(event, response):
     line, data = response.split('\n', 1)
     headers = dict([ x.split(':') for x in line.split() ])
-    print '%s %s | %s'%(headers['processname'], headers['channel'], data),
+    lines = data.split('\n')
+    prefix = '%s %s | '%(headers['processname'], headers['channel'])
+    print('\n'.join([ prefix + l for l in lines ]))
 
 if __name__ == '__main__':
     main()
